### PR TITLE
Added More Trace Attributes!

### PR DESCRIPTION
**Pull request by kdiluca**

### CODE REVIEW

File: src/thor/trace_attributes_action.cc [hunk 1]
```diff
@@ -26,37 +26,164 @@ namespace {
...
         const auto& edge = trip_path.node(i - 1).edge();
         auto names = json::array({});
 
         auto edgemap = json::map({});
+        if (edge.has_truck_route())
+          edgemap->emplace("truck_route", static_cast<bool>(edge.truck_route()));
+        if (edge.has_truck_speed())
+          edgemap->emplace("truck_speed", json::fp_t{edge.truck_speed(), 3});
+        if (edge.has_speed_limit())
+          edgemap->emplace("speed_limit", static_cast<uint64_t>(edge.speed_limit()));
```
Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: we will need to round

Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: and use scale

File: src/thor/trace_attributes_action.cc [hunk 2]
```diff
@@ -26,37 +26,164 @@ namespace {
...
+          edgemap->emplace("begin_shape_index", static_cast<uint64_t>(edge.begin_shape_index()));
+        }
+        if (edge.has_end_heading())
+          edgemap->emplace("end_heading", static_cast<uint64_t>(edge.end_heading()));
+        if (edge.has_begin_heading())
+          edgemap->emplace("begin_heading", static_cast<uint64_t>(edge.begin_heading()));
+      //  if (edge.has_road_class())
+      //    edgemap->emplace("road_class", edge.road_class());
+        if (edge.has_speed())
+          edgemap->emplace("speed", json::fp_t{edge.speed() * scale, 3});
```
Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: round (speed * scale)

File: src/thor/trace_attributes_action.cc [hunk 3]
```diff
@@ -83,13 +216,9 @@ namespace {
...
       {"edges", edges}
     });
-
-    // Add if it exists
     if (id)
       json->emplace("id", *id);
-
-    // Add shape if it exists
-    if (trip_path.has_shape())
+    if (returnShape)
```
Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: this needs to be changed back to this:
 if (trip_path.has_shape())
shape and shape indexes are separate flags

File: src/thor/trace_attributes_action.cc [hunk 4]
```diff
@@ -39,24 +87,101 @@ namespace {
         auto names = json::array({});
 
         auto edgemap = json::map({});
+        if (edge.has_truck_route())
+          edgemap->emplace("truck_route", static_cast<bool>(edge.truck_route()));
+        if (edge.has_truck_speed())
+          edgemap->emplace("truck_speed", json::fp_t{std::round(edge.truck_speed() * scale), 3});
```
Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: maybe just round speed_limit

Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: and no others

Comment (gknisely) [inline src/thor/trace_attributes_action.cc]: no all are rounded if mph exists on the data side

Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: round all speeds

**Comment by dgearhart** [comment]
:shipit: with some known TODOs

**Pull request merged**